consensus/hotstuff/integration/liveness_test.go

```diff
@@ -217,3 +218,69 @@ func Test1TimeoutOutof5Instances(t *testing.T) {
 		assert.Equal(t, finalizedViews, FinalizedViews(instances[i]), "instance %d should have same finalized view as first instance")
 	}
 }
+
+// TestBlockDelayIsHigherThanTimeout tests an edge case where block delay is the same as round duration
+// which results in timing edge cases where some nodes generate timeout objects but don't yet form a TC but then receive
+// a proposal with newer QC.
+func TestBlockDelayIsHigherThanTimeout(t *testing.T) {
```

Comment:
I guess it could be a problem, I will modify it so byzantine replicas drop all timeouts, incoming and outgoing

Reply:
Strictly speaking for this test it's not an issue because healthy replicas will never construct a TC and eventually they will construct proposal with valid QC, so even if blocked nodes have moved forward after constructing TC, they still have already contributed into creation of QC by sending votes. For sake of clarity I will block incoming TO as well though.